staging/src/k8s.io/apiserver/pkg/storage/cacher/ready.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cacher
+
+import (
+	"fmt"
+	"sync"
+)
+
+// ready is a condition variable that blocks until is ready if is not stopped
+type ready struct {
+	ok      bool
+	stopped bool // if stopped is true ok must be false
```

Comment:
non-blocking, but having two vars to reason about with a constraint between them seems tricker than a single state var that is one of NotReady, Ready, or Stopped.

Reply:
done, tri-state it is  better and more correct